rclcpp/include/rclcpp/function_traits.hpp

```diff
@@ -92,6 +92,18 @@ struct function_traits<
   : function_traits<ReturnTypeT(Args ...)>
 {};
 
+// std::bind for free functions
+template<typename ReturnTypeT, typename ... Args, typename ... FArgs>
+#if defined _LIBCPP_VERSION  // libc++ (Clang)
```

Comment:
I'm not so sure about this approach because it restricts our portability between compilers. The return type of std::bind doesn't have the same name in the implementation for these three compilers because it's not specified in the C++11 standard, right? That implies to me that we should try to come up with an approach that conforms better to the standard. I know we said we're only supporting these three major compilers, but it still feels wrong to me.